### ISSUE TITLE
az sql mi create crashes

### ISSUE THREAD

**Issue by kim0** [Azure/azure-cli#9510]
## Describe the bug
az sql mi create with certain options, causes python to barf

**Command Name**
`az sql mi create`

**Errors:**
```
The command failed with an unexpected error. Here is the traceback:



Traceback (most recent call last):
  File "/opt/az/lib/python3.6/site-packages/knack/cli.py", line 206, in invoke
    cmd_result = self.invocation.execute(args)
  File "/opt/az/lib/python3.6/site-packages/azure/cli/core/commands/__init__.py", line 560, in execute
    raise ex
  File "/opt/az/lib/python3.6/site-packages/azure/cli/core/commands/__init__.py", line 618, in _run_jobs_serially
    results.append(self._run_job(expanded_arg, cmd_copy))
  File "/opt/az/lib/python3.6/site-packages/azure/cli/core/commands/__init__.py", line 611, in _run_job
    six.reraise(*sys.exc_info())
  File "/opt/az/lib/python3.6/site-packages/six.py", line 693, in reraise
    raise value
  File "/opt/az/lib/python3.6/site-packages/azure/cli/core/commands/__init__.py", line 588, in _run_job
    result = cmd_copy(params)
  File "/opt/az/lib/python3.6/site-packages/azure/cli/core/commands/__init__.py", line 297, in __call__
    return self.handler(*args, **kwargs)
  File "/opt/az/lib/python3.6/site-packages/azure/cli/core/__init__.py", line 461, in default_command_handler
    return op(**command_args)
  File "/opt/az/lib/python3.6/site-packages/azure/cli/command_modules/sql/custom.py", line 2005, in managed_instance_create
    kwargs['sku'] = _find_managed_instance_sku_from_capabilities(cmd.cli_ctx, kwargs['location'], sku)
  File "/opt/az/lib/python3.6/site-packages/azure/cli/command_modules/sql/custom.py", line 1974, in _find_managed_instance_sku_from_capabilities
    managed_instance_version_capability = _get_default_capability(capabilities.supported_managed_instance_versions)
  File "/opt/az/lib/python3.6/site-packages/azure/cli/command_modules/sql/custom.py", line 118, in _get_default_capability
    next(c for c in capabilities if c.status == CapabilityStatus.available))
StopIteration
```

## To Reproduce:
Steps to reproduce the behavior. Note that argument values have been redacted, as they may contain sensitive information.

- _Put any pre-requisite steps here..._
- `az sql mi create -g {} -n {} -l {} -i -u {} -p {} --subnet {} --capacity {} --storage {} --edition {} --family {}`

## Environment Summary
```
Linux-4.15.0-1045-azure-x86_64-with-debian-stretch-sid
Python 3.6.5
Shell: bash

azure-cli 2.0.65

Extensions:
azure-devops 0.8.0
aks-preview 0.4.2

```

**Comment by jaredmoo** [comment]
This error is happening because Managed Instance is not available to your subscription in the specified region. I will improve the error message.

**Comment by jaredmoo** [comment]
Fixed by #9918
